sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -246,7 +247,12 @@ case class DataSource(
       case s: StreamSinkProvider =>
         s.createSink(sparkSession.sqlContext, options, partitionColumns, outputMode)
 
-      case parquet: parquet.ParquetFileFormat =>
+      // TODO: Remove the `isInstanceOf` check when other formats have been ported
+      case fileFormat: FileFormat
+        if (fileFormat.isInstanceOf[CSVFileFormat]
+          || fileFormat.isInstanceOf[JsonFileFormat]
```

Comment:
@ScrapCodes , thanks! But I'm afraid that syntax would raise a compilation error:
```
[ERROR] .../datasources/DataSource.scala:250: illegal variable in pattern alternative
[ERROR]   case fileFormat: CSVFileFormat | JsonFileFormat | ParquetFileFormat | TextFileFormat =>
[ERROR]        ^
```
A work-around can be the following, but I found it somewhat less intuitive:
```scala
case fileFormat@(_: CSVFileFormat |
                 _: JsonFileFormat |
                 _: ParquetFileFormat |
                 _: TextFileFormat) =>
  // other code
  ... fileFormat.asInstanceOf[FileFormat] ...
```